src/Storages/MergeTree/MergeTreeDataPartWriterOnDisk.h

```diff
@@ -71,42 +100,46 @@ class MergeTreeDataPartWriterOnDisk : public IMergeTreeDataPartWriter
         const MergeTreeWriterSettings & settings,
         const MergeTreeIndexGranularity & index_granularity);
 
-    void calculateAndSerializePrimaryIndex(const Block & primary_index_block) final;
-    void calculateAndSerializeSkipIndices(const Block & skip_indexes_block) final;
-
-     /// Count index_granularity for block and store in `index_granularity`
-    size_t computeIndexGranularity(const Block & block);
-    virtual void fillIndexGranularity(size_t index_granularity_for_block, size_t rows_in_block);
-
-    void initSkipIndices() final;
-    void initPrimaryIndex() final;
-
-    void finishPrimaryIndexSerialization(MergeTreeData::DataPart::Checksums & checksums, bool sync) final;
-    void finishSkipIndicesSerialization(MergeTreeData::DataPart::Checksums & checksums, bool sync) final;
-
     void setWrittenOffsetColumns(WrittenOffsetColumns * written_offset_columns_)
     {
         written_offset_columns = written_offset_columns_;
     }
 
 protected:
-    using SerializationState = IDataType::SerializeBinaryBulkStatePtr;
-    using SerializationStates = std::unordered_map<String, SerializationState>;
+     /// Count index_granularity for block and store in `index_granularity`
+    size_t computeIndexGranularity(const Block & block) const;
```

Comment:
By the way, `index granularity` was somewhat confusing to me now that we have secondary indexes. Maybe `order_by_granularity` or something would be better...

Reply:
Actually, `order_by_granularity` is more confusing because of not all columns from `ORDER BY` expression stored in `primary.idx` on disk. Secondary indices granularity calculated in primary index granules. I agree that these interchangeable words like `mark`, `granularity`, `rows in mark`, `index granularity` and so on are very confusing at first glance.